package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -150,8 +152,28 @@ def parse(self, **kwargs):
         charges = []
 
         for a in atom_lines:
-            aid, name, x, y, z, atom_type, resid, resname, charge = a.split()[:9]
-
+            columns = a.split()
+            if len(columns) >= 9:
+                columns = columns[:9]
+                aid, name, x, y, z, atom_type, resid, resname, charge = columns
+            else:
+                opt_values = ['1', 'UNK', '0.0']
+                opt_fields = ['subst_id', 'subst_name', 'charge']
+                if len(columns) < 6:
+                    raise ValueError("The @<TRIPOS>ATOM block in mol2 file {0}"
+                                     " should contain at least 6 fields to be"
+                                     " unpacked: atom_id atom_name x y z"
+                                     " atom_type [subst_id[subst_name"
+                                     " [charge [status_bit]]]]".format(
+                                         os.path.basename(self.filename)))
+                aid, name, x, y, z, atom_type = columns[:6]
+                for i in range(6, len(columns)):
+                    opt_values[i-6] = columns[i]
+                for i in range(len(columns), 9):
+                    warnings.warn("Not enough values to unpack."
+                                  " {0} has been given value {1}.".format(
+                                      opt_fields[i-6], opt_values[i-6]))
```

Comment:
See comment above about discarding attributes instead of providing arbitrary default values. If the MOL2 file is standard-compliant it should not throw warnings.